fuel-txpool/src/containers/dependency.rs

```diff
@@ -98,6 +101,33 @@ impl Dependency {
         }
     }
 
+    fn check_if_coin_input_can_spend_db_coin(coin: &Coin, input: &Input) -> anyhow::Result<()> {
+        match input {
+            Input::Coin {
+                utxo_id,
+                owner,
+                amount,
+                asset_id,
+                ..
+            } => {
+                if *owner != coin.owner {
+                    return Err(Error::NotInsertedIoWrongOwner.into());
+                }
+                if *amount != coin.amount {
+                    return Err(Error::NotInsertedIoWrongAmount.into());
+                }
+                if *asset_id != coin.asset_id {
+                    return Err(Error::NotInsertedIoWrongAssetId.into());
+                }
+                if coin.status == CoinStatus::Spent {
+                    return Err(Error::NotInsertedInputUtxoIdSpent(*utxo_id).into());
+                }
+                Ok(())
+            }
+            _ => panic!("Use it only for coin output check"),
```

Comment:
why not just leverage the ability to use a result here?

```suggestion
            _ => Err(anyhow!("Use it only for coin output check")),
```

Reply:
Function is missused if it comes to that point, and this is helper internal function so i was fine to just panic. Will change it to error